packages/catalog-model/src/location/helpers.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from './types';
+
+/**
+ * Parses a string form location reference.
+ *
+ * @param ref A string-form location reference, e.g. 'url:https://host'
+ * @returns A location reference, e.g. { type: 'url', target: 'https://host' }
+ */
+export function parseLocationReference(ref: string): LocationSpec {
+  if (typeof ref !== 'string') {
+    throw new TypeError(
+      `Unable to parse location reference '${ref}', unexpected argument ${typeof ref}`,
+    );
+  }
+
+  const splitIndex = ref.indexOf(':');
+  if (splitIndex === 0) {
+    throw new TypeError(
+      `Unable to parse location reference '${ref}', empty type`,
+    );
+  } else if (splitIndex === -1) {
+    return { type: ref, target: '' };
```

Comment:
This seems strange, `github.com` would be parsed as `{type: 'github.com', target: ''}`? Would assume we'd just want to throw in this case

Reply:
I though of being able to just say `bootstrap` for example. Poor example, since it's going away, but let's say `sysmodel` then for illustration. It will be pretty common to just want to trigger org specific processors A, B, C and D. So a bit wasteful to force the extra target then

Reply:
Would probably still be `sysmodel:https://sysmodel.internal-stuff.spotify.com` or something of that sort right? Not quite seeing the point to integrate something using a location type processor and then configure it separately. Perhaps a `sysmodel` think could be a location provider too? :>

Reply:
Lookup isn't necessarily on url form. I have brought up this before that there's a frustrating need to have "trigger locations" seemingly unnecessarily, and that one often would essentially like to put config on locations - or to do away with the location entirely.

Sure maybe we can make them location providers, that could be neat. But then they would have to assign themselves an effective, stable, unique location internally since things are keyed on that

Reply:
Part of the point of location providers is that they could be a root source, and locations produced by them don't need a parent, which is how we'd get rid of the bootstrap location

Reply:
Right, but what they output is still locations. So if this one would be self-supplying, it would still return a type/target pair, whose values it needs to "make up"

Reply:
You mean a sysmodel location provider that emits a sysmodel location, which is then picked up by a sysmodel location processor to then output the actual location? O_o

Reply:
Yes, roughly. Given that we want to make the processing parallelizable, I was under the impression that our central state table would be keyed on a location ref string. Hosts will want to track and lock the sysmodel fetching as well.

Reply:
(made it throw)